src/plugins.ts

```diff
@@ -124,15 +133,24 @@ export default class Plugins {
     root: string,
     {prod = true}: { prod?: boolean } = {},
   ): Promise<void> {
-    if (fs.existsSync(path.join(root, 'yarn.lock'))) {
-      this.debug(`yarn.lock exists at ${root}. Installing prod dependencies`)
+    const doRefresh = async () => {
       // use yarn.lock to fetch dependencies
       await this.yarn.exec(prod ? ['--prod'] : [], {
         cwd: root,
         verbose: this.verbose,
       })
+    }
+
+    if (await fileExists(path.join(root, 'oclif.lock'))) {
+      this.debug(`oclif.lock exists at ${root}. Installing prod dependencies`)
+      await fs.promises.rename(path.join(root, 'oclif.lock'), path.join(root, 'yarn.lock'))
+      await doRefresh()
+      await fs.promises.unlink(path.join(root, 'yarn.lock'))
+    } else if (await fileExists(path.join(root, 'yarn.lock'))) {
```

Comment:
Should we check for `yarn.lock` first? Say `npm@10` comes out and it allows you to package a `yarn.lock`. I could see a situation where someone has an old `oclif.lock` checked into their repo (even though we discourage that [here](https://github.com/oclif/oclif/pull/1178/files#diff-564b0ff4135122f20b60beb380edcc5b86918c3d258ae31d6d56b4f3f7b0d4c1R17)) and the shipped `yarn.lock` would get overwritten with an old `oclif.lock`.